### ISSUE TITLE
Removing seeds does not work

### ISSUE THREAD

**Issue by JTrenerry** [UQcsse3200/2023-studio-1#523]
## Description
When placing a seed on a plant / seedling / seed (any plant on a croptile) the seed is consumed but not actually used to plant anything. This locks you out of the entire game if done in the tutorial.

## How To Reproduce
Steps to reproduce the behaviour:
1. Get 2 seeds
2. Hoe the ground
3. Plant one on the tilled ground
4. Plant one on top of the other

## Expected Behaviour
One seed is consumed and the other is not.

## Additional Context
This can force the player to reset as they would be left without seeds and thus needs to be fixed

## Temporary Fix
Do not plant a seed on another plant

**Comment by JTrenerry** [comment]
This may also impact fertilizer depending on if it should be removed when the ground is already fertilized.

**Comment by JTrenerry** [comment]
Solution for both is to check the tile and have an accurate bool return value for both fertilizer and planting.

**Comment by 86LAK** [comment]
🦖

**Comment by Tom-Strooper** [comment]
This bug is caused by a double implementation of the same feature. PR #490 introduced the behavior of removing consumable items on use. PR #502 also introduced the same feature and was merged into the main 2 hours later, without removing the code from #490, using the `perishable` boolean value to determine whether items should be removed. However, this implementation removes items from the inventory regardless of whether the item was actually used, hence the bug. I have added team 8 to this bug as PR #502 was introduced by team 8.
